.github/workflows/README.md

```diff
@@ -0,0 +1,50 @@
+## CI/CD
+
+There are 5 main stages that run on GitHub actions:
+
+* Build
+* Unit Test
+* Integration Test
+* Release
+
+There are some other stages that run for every push on the main branches:
+
+* [Microbenchmark](./microbenchmark.yml)
+* [Snapshoty](./snapshoty.yml)
+* [Sync-branches](./sync-branches.yml)
+
+### Scenarios
+
+* Tests should be triggered on branch, tag and PR basis.
+* Commits that are only affecting the docs files should not trigger any test or similar stages that are not required.
+* Automated release in the CI gets triggered through a GitHub workflow.
+* **This is not the case yet**, but if Github secrets are required then Pull Requests from forked repositories won't run any build accessing those secrets. If needed, then create a feature branch.
+
+### How to interact with the CI?
+
+#### On a PR basis
+
+Once a PR has been opened then there are two different ways you can trigger builds in the CI:
+
+1. Commit based
+1. UI based, any Elasticians can force a build through the GitHub UI
```

Comment:
Could you elaborate on those ?
- commit based: is this when pushing to a branch ? If yes, then just adding it for extra clarity would be great.
- UI based: is that by manually triggering the build in GH actions ? Do we still have the comment-based triggers available ? (I'm not asking for adding those if it's not the case anymore, I'd just like we document them if they are already there).

Reply:
> commit based: is this when pushing to a branch ? If yes, then just adding it for extra clarity would be great.

When a commit has been pushed to the upstream (elastic/apm-agent-java) or the origin when a Pull Request has been created previously.

What do you think if to change it to `Git commit based`?

>  that by manually triggering the build in GH actions 

Correct

> Do we still have the comment-based triggers available ?

No, they are not supported yet. It requires to use GitHub review comments rather than GitHub comments so those builds runs per a given commit rather than the latest, for security reasons